demos/common/python/models/detection_model.py

```diff
@@ -0,0 +1,76 @@
+from .image_model import ImageModel
+from .utils import Detection, load_labels, clip_detections
+
+
+class DetectionModel(ImageModel):
+    def __init__(self, ie, model_path, input_transform=None, resize_type='default',
+                 labels=None, threshold=None, iou_threshold=None):
+        super().__init__(ie, model_path, input_transform=input_transform, resize_type=resize_type)
+        if isinstance(labels, (list, tuple)):
+            self.labels = labels
+        else:
+            self.labels = load_labels(labels) if labels else None
+
+        self.threshold = threshold
+        self.iou_threshold = iou_threshold
+
+    def postprocess(self, outputs, meta):
+        detections = self._parse_outputs(outputs, meta)
+        resized_shape = meta['resized_shape']
+        original_shape = meta['original_shape']
+
+        if self.resize_type=='letterbox':
+            detections = self._resize_detections_letterbox(detections, original_shape[1::-1], resized_shape[1::-1])
+        elif self.resize_type == 'keep_aspect_ratio':
+            detections = self._resize_detections_with_aspect(detections, original_shape[1::-1], resized_shape[1::-1], (self.w, self.h))
+        elif self.resize_type == 'default':
+            detections = self._resize_detections(detections, original_shape[1::-1])
+        else:
+            raise RuntimeError('Unknown resize type {}'.format(self.resize_type))
```

Comment:
I would like to incapsulate approach selection logic in _resize_detections method

Reply:
Done